docs/writing-tests/making-a-testing-plan.md

```diff
@@ -0,0 +1,507 @@
+  features
+- web developers often want to test discrepancies between browsers on their web
+  applications
+
+(If you don't have any particular goal, we can help you get started. Check out
+[the issues labeled with `type:missing-coverage` on
+GitHub.com](https://github.com/web-platform-tests/wpt/labels/type%3Amissing-coverage).
+Leave a comment if you'd like to get started with one, and don't hesitate to
+ask clarifying questions!)
+
+This guide will:
+
+1. show you how to use the specifications to learn what kinds of tests will be
+   most helpful
+2. help you develop a sense for what *doesn't* need to be tested
+3. demonstrate methods for figuring out which tests (if any) have already been
+   written for WPT
+
+## Understanding the "testing surface"
+
+Web platform specifications are instructions about how a feature should work.
+They're critical for implementers to "build the right thing," but they are also
+important for anyone writing tests. We can use the same instructions to infer
+what kinds of tests would be likely to detect mistakes. Here are a few common
+patterns in specification text and the kind of tests they suggest.
+
+### Input sources
+
+Algorithms may accept input from many sources. Modifying the input is the most
+direct way we can influence the browser's behavior and verify that it matches
+the specifications. That's why it's helpful to be able to recognize all the
+different sources of input.
+
+```eval_rst
+================ ==============================================================
+Type of feature  Potential input sources
+================ ==============================================================
+JavaScript       parameters, `context object <https://dom.spec.whatwg.org/#context-object>`_
+HTML             element content, attributes, attribute values
+CSS              selector strings, property values, markup
+================ ==============================================================
+```
+
+*Example:* This is the first step of the `Notification` constructor from [the
+Notifications standard](https://notifications.spec.whatwg.org/#constructors):
+
+> The Notification(title, options) constructor, when invoked, must run these steps:
+>
+> 1. If the [current global
+>    object](https://html.spec.whatwg.org/multipage/webappapis.html#current-global-object)
+>    is a
+>    [ServiceWorkerGlobalScope](https://w3c.github.io/ServiceWorker/#serviceworkerglobalscope)
+>    object, then [throw](https://heycam.github.io/webidl/#dfn-throw) a
+>    `TypeError` exception.
+> 2. Let *notification* be the result of [creating a
+>    notification](https://notifications.spec.whatwg.org/#create-a-notification)
+>    given *title* and *options*. Rethrow any exceptions.
+>
+> [...]
+
+A thorough test suite for this constructor will include tests for the behavior
+of many different values of the *title* parameter and the *options* parameter.
+Choosing those values is a challenge unto itself--see [Avoid Excessive
```

Comment:
```suggestion
Choosing those values is a challenge unto itself - see [Avoid Excessive
```

The '--' seems to render weirdly in the markdown; should this just be a single hyphen?

Reply:
[Philip also noticed this](https://github.com/web-platform-tests/wpt/pull/19628#discussion_r336737090), but since the build we're targeting (the Python Sphinx tool) automatically creates em dashes for this pattern, we're keeping it.